business/services.go

```diff
@@ -153,6 +153,14 @@ func (in *SvcService) getServiceListForCluster(ctx context.Context, criteria Ser
 			log.Errorf("Error fetching Services per namespace %s: %s", criteria.Namespace, err2)
 			errChan <- err2
 		}
+		// when services were not loaded, try once more for remote clusters only by filtering services by selector
+		if len(svcs) == 0 && len(selectorLabels) != 0 && cluster != in.config.KubernetesConfig.ClusterName {
+			svcs, err2 = kubeCache.GetAllServicesFilteredBySelector(criteria.Namespace, selectorLabels)
```

Comment:
Sorry what's confusing me here is what does `ServiceCriteria.ServiceSelector` mean? On every other `Get<Objects>` method the selector selects the labels of the object and not the `Spec.Selector` like `GetPods` which returns all pods that match the passed in label selector. It seems like all callers of `GetServices` actually want the `service.Spec.Selector` from where I can tell it's used in the workloads and apps services. Given that, I think what you had in the previous commit was fine and it's better than doing multiple calls here. Can you simply revert to the previous commit and add a comment to the `GetServices` method that when you pass `selectorLabels` you are actually filtering on `Spec.Selector` rather than `Metadata.Labels`?

Reply:
ok it's clear now, have reverted back